roles/lib_openshift/src/doc/certificate_authority

```diff
@@ -0,0 +1,141 @@
+    - When create-key-pair is desired then the following parameters are passed.
+    - ['private_key', 'public_key']
+    - When create-server-cert is desired then the following parameters are passed.
+    - ['cert', 'key', 'signer_cert', 'signer_key', 'signer_serial']
+    required: false
+    default: present
+    choices: 
+    - present
+    aliases: []
+  kubeconfig:
+    description:
+    - The path for the kubeconfig file to use for authentication
+    required: false
+    default: /etc/origin/master/admin.kubeconfig
+    aliases: []
+  debug:
+    description:
+    - Turn on debug output.
+    required: false
+    default: False
+    aliases: []
+  cmd:
+    description:
+    - The sub command given for `oc adm ca`
+    required: false
+    default: None
+    choices:
+    - create-master-certs
+    - create-key-pair
+    - create-server-cert
+    aliases: []
+  cert_dir:
+    description:
+    - The certificate data directory.
+    required: false
+    default: None
+    aliases: []
+  cert:
+    description:
+    - The certificate file. Choose a name that indicates what the service is.
+    required: false
+    default: None
+    aliases: []
+  key:
+    description:
+    - The key file. Choose a name that indicates what the service is.
+    required: false
+    default: None
+    aliases: []
+  overwrite:
+    description:
+    - Overwrite existing cert files if found.  If false, any existing file will be left as-is.
+    required: false
+    default: False
+    aliases: []
+  signer_cert:
+    description:
+    - The signer certificate file.
+    required: false
+    default: None
+    aliases: []
+  signer_key:
+    description:
+    - The signer key file.
+    required: false
+    default: None
+    aliases: []
+  signer_serial:
+    description:
+    - The signer serial file.
+    required: false
+    default: None
+    aliases: []
+  public_key:
+    description:
+    - The public key file used with create-key-pair
+    required: false
+    default: None
+    aliases: []
+  private_key:
+    description:
+    - The private key file used with create-key-pair
+    required: false
+    default: None
+    aliases: []
+    
+  hostnames:
+    description:
+    - Every hostname or IP that server certs should be valid for (comma-delimited list)
+    required: false
+    default: None
+    aliases: []
+  master:
+    description:
+    - The API server's URL
+    required: false
+    default: None
+    aliases: []
+  public_master:
+    description:
+    - The API public facing server's URL (if applicable)
+    required: false
+    default: None
+    aliases: []
+  signer_name:
+    description:
+    - The name to use for the generated signer
+    required: false
+    default: None
+    aliases: []
+author:
+- "Kenny Woodson <kwoodson@redhat.com>"
+extends_documentation_fragment: []
+'''
+
+EXAMPLES = '''
+- name: Create a self-signed cert
+  oadm_ca:
+    cmd: create-server-cert
```

Comment:
Out of curiosity, is there a particular use case that is driving the porting of this module now? It doesn't seem like replacing the current calls to generate certificates (that interact with the oc tooling) would buy us much if any improvements to idempotency.

I also know that discussions with @abutcher seemed to indicate that there is going to be a new cert workflow coming that would allow us to do all certificates through the OpenShift CA as opposed to only the subset we are able to do now.

If this is a dependency of the router and registry modules, would it be be better to modify them to accept a pre-created certificates instead as a stopgap?